File: tests/fixtures/java/MatchSwitch.java
// Expected metrics:
// categorize: cognitive=3, cyclomatic=6, sloc=20

class MatchSwitch {
    int categorize(int code) {
        int result;
        switch (code) {
            case 1:
                result = 10;
                break;
            case 2:
                result = 20;
                break;
            case 3:
                if (code > 0) {
                    result = 30;
                }
                break;
            default:
                result = -1;
                break;
        }
        return result;
    }
}

File: tests/fixtures/rust/match_switch.rs
// Expected metrics:
// categorize: cognitive=4, cyclomatic=6, sloc=14
fn categorize(x: i32) -> &'static str {
    match x {
        1 => "one",
        2 => {
            if x > 0 {
                "positive two"
            } else {
                "two"
            }
        }
        3 => "three",
        _ => "other",
    }
}
